ReactAndroid/src/main/java/com/facebook/react/modules/core/DeviceEventManagerModule.java

```diff
@@ -53,6 +56,15 @@ public void emitHardwareBackPressed() {
   }
 
   /**
+   * Sends an event to the JS instance that a new intent was received.
+   */
+  public void emitNewIntent(Uri uri) {
+    getReactApplicationContext()
+        .getJSModule(RCTDeviceEventEmitter.class)
+        .emit("openURL", uri.toString());
```

Comment:
Yeah, as @satya164 said, it felt like there should be an intermediate layer (ActivityEventListener?) to which it sends the intent, and then that module makes the choice to interpret it as an opened URL, but that pathway wasn't quite clear.